pkg/zonemgr/zone_manager.go

```diff
@@ -27,19 +28,27 @@ type ZoneManager struct {
 	zoneFile      *ZoneFile
 }
 
-func NewZoneManager() *ZoneManager {
+func NewZoneManager() (*ZoneManager, error) {
 	zoneMgr := &ZoneManager{}
-	zoneMgr.prepare()
-	return zoneMgr
+	err := zoneMgr.prepare()
+	return zoneMgr, err
 }
 
-func (zoneMgr *ZoneManager) prepare() {
-	domain := os.Getenv(envVarDomain)
+func (zoneMgr *ZoneManager) prepare() error {
+	domain := domainDefault
 	nameServerIP := os.Getenv(envVarNameServerIP)
+	if customDomain := os.Getenv(envVarDomain); customDomain != "" {
+		domain = fmt.Sprintf("%s.%s", domain, customDomain)
+	}
+	zoneFileName := zoneFileNamePrefix + domain
+	zoneMgr.zoneFile = NewZoneFile(zoneFileName)
 
-	zoneMgr.zoneFileCache = NewZoneFileCache(nameServerIP, domain)
-
-	zoneMgr.zoneFile = NewZoneFile(zoneFileName + zoneMgr.zoneFileCache.domain)
+	if soaSerial, err := zoneMgr.zoneFile.ReadSoaSerial(); err != nil {
+		return err
+	} else {
```

Comment:
the else can be dropped, please simply do the code in it, because the if caught it had a return

Reply:
changed following https://github.com/kubevirt/kubesecondarydns/pull/45#discussion_r1054306875